### PR TITLE
feat(kind): Add telemetry to see what is happening

### LINKED ISSUE #1954: Telemetry for Kind
We should be able to know:
- ~~If the extension is enabled/disabled~~ it's not specific to kind and depends on https://github.com/containers/podman-desktop/issues/268 (I've added comment there)
- [ ] When users are creating, ~~deleting, stopping~~ a kind cluster
- ~~When users are doing "Play Kubernetes YAML" on Kind cluster~~ (invalid)
- ~~When people are using the CLI within Podman Desktop over a kind cluster~~ (not possible technically)

related also:
- #1909

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
Send some telemetry events about kind support (success/error)
like: click on Download kind, accept or not, succeed or not
create cluster with provider being selected, if there is success or not

depends on 
-  https://github.com/containers/podman-desktop/pull/1996

Also log usage of create clusters

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

fixes https://github.com/containers/podman-desktop/issues/1954

### How to test this PR?

Look at unit tests
Look at segment debugger

Change-Id: I7f7a59360e5ed32edc3a6b4d4593c9771928705e

### CODE REVIEW

**Review by lstocchi — APPROVED** [review]
I don't have access to the segment debugger but the code looks good 👍

File: extensions/kind/src/create-cluster.ts [hunk 1]
```diff
@@ -79,13 +80,18 @@ export async function createCluster(
...
   const result = await runCliCommand(kindCli, ['create', 'cluster', '--config', tmpFilePath], { env, logger });
 
   // delete temporary directory/file
   await fs.promises.rm(tmpDirectory, { recursive: true });
 
   if (result.exitCode !== 0) {
+    telemetryLogger.logError('createCluster', { provider, error: result.error, stdErr: result.stdErr });
     throw new Error(`Failed to create kind cluster. ${result.error}`);
+  } else {
+    telemetryLogger.logUsage('createClusterSuccess', { provider });
```
Comment (deboer-tim) [inline extensions/kind/src/create-cluster.ts]: Why send a separate event for createCluster and createClusterSuccess? Then you need to correlate to see success %age. I was expecting a single post-event with e.g. { provider, httpHostPort, httpsHostPort, success/fail boolean }. (and still logging error if applicable)

Comment (benoitf) [reply extensions/kind/src/create-cluster.ts]: logging the error is on the logError callback

Comment (deboer-tim) [reply extensions/kind/src/create-cluster.ts]: Yes, but today there are two events in either path (create + success vs create + error) and if you want to check something like success rate you need to correlate two types of events. In the future if we wanted to do something like adding an attribute for how long it takes we'd have to log it in both success and error events and analyze these separately.

Always sending the createCluster after the action with an extra boolean would mean one less event on the success path (post-create vs post-create + error) and you'd only need to look at the one event unless you were interested in failures.

Comment (benoitf) [reply extensions/kind/src/create-cluster.ts]: I've updated to a single event (with error or usage)

**Pull request merged**